include/avif/internal.h

```diff
@@ -59,6 +59,17 @@ static inline void avifBreakOnError()
         }                                 \
     } while (0)
 
+// Same as AVIF_CHECKERR() but also assert() in debug build configuration.
+// Can be used instead of assert() for extra security in release builds.
+#define AVIF_ASSERT_OR_RETURN(A, ERR) \
+    do {                              \
+        if (!(A)) {                   \
+            assert(AVIF_FALSE);       \
+            avifBreakOnError();       \
+            return ERR;               \
+        }                             \
+    } while (0)
```

Comment:
Since `assert()` is documented to depend on the `NDEBUG` macro:
https://en.cppreference.com/w/c/error/assert

I suggest we define `AVIF_ASSERT_OR_RETURN` as follows:

```
#ifdef NDEBUG
#define AVIF_ASSERT_OR_RETURN(A) AVIF_CHECKERR(A, AVIF_RESULT_INTERNAL_ERROR)
#else
#define AVIF_ASSERT_OR_RETURN(A) assert(A)
#endif
```